eduNEXT/platform-plugin-turnitin
# feat: add ora submission view filter for turnitin integration

### Description
This PR adds a pipeline step that modifies the ORA submission view rendering process. This step changes the default template by a custom template that has a warning message that notifies the student the Turnitin plagiarism check is enabled.


### How to Test

1. Install this plugin with this branch in the LMS/CMS services.
2. Install `openedx-filters` in the LMS/CMS services with [these changes](https://github.com/eduNEXT/openedx-filters/pull/1)
3. Install `edx-ora2` in the LMS/CMS services with [these changes](https://github.com/eduNEXT/edx-ora2/pull/13)
4. Add the Open edX Filters configuration to your environment. You can use this inline plugin:

    ```yaml
    name: filters-settings
    version: 0.1.0
    patches:
      openedx-common-settings: |
        OPEN_EDX_FILTERS_CONFIG = {
          "org.openedx.learning.ora.submission_view.render.started.v1": {
            "fail_silently": False,
            "pipeline": [
              "platform_plugin_turnitin.extensions.filters.ORASubmissionViewTurnitinWarning",
            ]
          },
        }
    ```
5. Restart services
6. In a unit of the course create a new ora assessment.
7. In the LMS, in the submission step (**1. Your Response**), you'll see:

    ![image](https://github.com/eduNEXT/platform-plugin-turnitin/assets/64033729/96786562-ecd1-4dc1-8821-f20eef108552)
